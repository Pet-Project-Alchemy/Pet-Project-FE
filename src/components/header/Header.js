import React from 'react';
import  './Header.scss';
import logo from '../assets/dogLogo.png';

const Header = () => {
  return (
    <header className='header'>
      <div className= 'header__logo-box'>
        <img className='header__logo' src={logo} alt='dog-log0' />
      </div>
      <div className='header__text-box'>
        <h1 className='heading__primary'>
          <span className='heading__primary-main'> Dog Out</span>
          <span className='heading__primary-sub'>outside</span>
        </h1>
      </div>
    </header>
  );
};

export default Header;
