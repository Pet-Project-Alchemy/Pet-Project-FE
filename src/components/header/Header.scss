@import '../abstract/variables';
@import '../abstract/base.scss';
@import '../abstract/animate.scss';

.header {
  height: 95vh;
  background-image: linear-gradient(to right bottom, rgba($color-primary-grey, 0.8), rgba($color-black, 0.8)),
    url('../assets/doghero.jpeg');
  background-size: cover;
  background-position: top;
  position: relative;
  clip-path: polygon(25% 0%, 100% 0%, 75% 100%, 0% 100%);
  
  &__text-box {
    position: absolute;
    top: 30%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
  }
}
.background {
  background-color: 
  $color-black;
}
.heading-primary {
  color: $color-primary-white;
  text-transform: uppercase;
  backface-visibility: hidden;
  margin-bottom: 4rem;
  
  &__sub--main {
    color: $color-primary-white;
    display: block;
    font-size: 4rem;
    font-weight: 400;
    letter-spacing: 2rem;
    animation: moveInLeft 1s ease-out;
  }
  &__sub {
    display: block;
    font-size: 2rem;
    font-weight: 700;
    letter-spacing: 1.75rem;
    animation: moveInRight 1s ease-in;
  }
}