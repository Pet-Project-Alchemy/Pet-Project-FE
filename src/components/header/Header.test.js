import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header component', () => {
  it('renders Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
