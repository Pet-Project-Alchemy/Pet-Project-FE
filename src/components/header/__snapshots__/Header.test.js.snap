// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<header
  className="header"
>
  <div
    className="header__logo-box"
  >
    <img
      alt="dog-log0"
      className="header__logo"
      src="/path/image"
    />
  </div>
  <div
    className="header__text-box"
  >
    <h1
      className="heading__primary"
    >
      <span
        className="heading__primary-main"
      >
         Dog Out
      </span>
      <span
        className="heading__primary-sub"
      >
        outside
      </span>
    </h1>
  </div>
</header>
`;
